File: Move_Zeros_To_End.java
package Array_Problems;

public class Move_Zeros_To_End {
    public static void main(String[] args) {
        int[] nums = {0,1,0,3,12};
        move(nums);
        for(int num:nums){
            System.out.print(num+" ");
        }
    }
    public static void move(int[] nums){
        int i=0;
        for(int j=0;j<nums.length;j++){
            if(nums[j]!=0){
                int temp=nums[j];
                nums[j]=nums[i];
                nums[i]=temp;
                i++;
            }
        }
    }
}

File: Move Zeros to end.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        count = 0
        result = []

        for num in nums:
            if num == 0:
                count += 1
            else:
                result.append(num)

        result.extend([0] * count)

        # Modify nums in-place
        nums[:] = result  # Copy result back to nums
